dali/operators/generic/slice/slice_attr.h

```diff
@@ -88,8 +113,16 @@ class SliceAttr {
       "Expected 3 inputs. Received: " + std::to_string(ws.NumInput()));
     const auto &crop_anchor = ws.Input<CPUBackend>(1);
     const auto &crop_shape = ws.Input<CPUBackend>(2);
+    DALI_ENFORCE(crop_anchor.type().id() == crop_shape.type().id(),
+                  make_string("Anchor and shape should have the same type. Got: ",
+                              crop_anchor.type().id(), " and ", crop_shape.type().id()));
+    auto args_dtype = crop_anchor.type().id();
     VerifyArgsShape(crop_anchor.shape(), crop_shape.shape());
-    ProcessArgumentsHelper(ws.data_idx(), crop_anchor.data<float>(), crop_shape.data<float>());
+    TYPE_SWITCH(args_dtype, type2id, ArgsType, SLICE_ARGS_TYPES, (
+      ProcessArgumentsHelper(ws.data_idx(),
+                             crop_anchor.data<ArgsType>(),
+                             crop_shape.data<ArgsType>());
+    ), DALI_FAIL(make_string("Unsupported type anchor and shape arguments: ", args_dtype)));  // NOLINT
```

Comment:
This part of the code repeats 3 times. Maybe you can extract it?

Reply:
It is hard because there are subtle changes because of the different workspace APIs. Do you have any particular ideas?

Reply:
You can use InputRef instead of Input, so you'll always get a TensorList or a TensorVector. Are there any other differences?